### ISSUE TITLE
Background images set with CSS do not uses the *.webp version if present

### ISSUE THREAD

**Issue by masteradhoc** [WordPress/performance#414]
## Bug Description
We already got a fix at #288 by @mitogh for the featured images - thanks for that! Some time ago i developed a small simple website with gutenberg & generatepress. i tried to use as many default blocks as possible. Some elements are working with the webp replacement as expected - some not yet.

generatepress offers a container block. it allows to choose a image from the library and sets its url into the field
![image](https://user-images.githubusercontent.com/6242098/176944046-d903c2fb-ab3b-499e-bec6-7905727e9bdc.png)

the background image is set over css and currently not replaced by a webp image - although one would be available
![image](https://user-images.githubusercontent.com/6242098/176944230-995a684a-fd1a-41ee-862f-c0d3928747b9.png)

This currently seems to be the only issue left on my testpage where webp images dont get automatically choosen.

## Steps to reproduce

1. Upload jpeg images - implement them as background images over css 
2. install performance plugin, activate webp
3. use Regenerate Thumbnails plugin to generate the web
4. See on frontend if they get used as webp or still jpg

## Additional Context

 - PHP Version: 8.0
 - OS: Windows
 - Browser: MS Edge
 - Plugin Version: Version 1.2.0

**Comment by eugene-manuilov** [comment]
Thanks for submitting this ticket, @masteradhoc, but unfortunately I don't think we can help with this use-case. There are multiple reasons why.

First of all, as you can see the image URL is saved as a custom field and there is no way to detect its attachment ID.  We can't replace a jpeg image with webp without knowing attachment ID because we need to know whether the image has webp variants. Also based on your screenshots, it seems that the image URL is added to the autogenerated stylesheet file which is also not under our control.

@adamsilverstein can you confirm it as well? Going to mark this ticket as `wontfix` for now.

**Comment by masteradhoc** [comment]
@eugene-manuilov Thanks. i'll check with the themeowner also if and how its exactly implemented and get back here.

What i also noticed is that even if you currently upload an image it will stay as jpg inside the file url field in media library.
With the current implementation this results for me in the usecase noted above that even rechoosing the file again over the media popup the link always will point to the jpg and not the webp image thats available.

--

EDIT: GeneratePress also allows the image to be added inline instead the way mentioned above. i tried it also but the images havent successfully been replaced by a webp. this is the code that the themeautor uses.
The function that adds the inline URL is here: https://github.com/tomusborne/generateblocks/blob/1.5.2/includes/general.php#L313
The function that fetches the image URL can be found here: https://github.com/tomusborne/generateblocks/blob/1.5.2/includes/functions.php#L513

Maybe helps to see the issue. available also on slack to check together.

**Comment by eugene-manuilov** [comment]
Thanks, @masteradhoc. I have created a PR #420 that should help with this issue. It adds a hook for the `wp_get_attachment_image_src` filter and updates an image URL to use webp alternative **if the function has been called in the `the_content` context** (doing `the_content` filter).

**Comment by felixarntz** [comment]
@masteradhoc @eugene-manuilov We plan to address this for the core implementation by adding a parameter to request a MIME type to various core image functions, and the plan is also to automatically use the WebP version by default in a frontend body context (see #379) if that parameter isn't passed.

However, because this ties deeply into the core image functions, I don't think it's reasonable to address this in the Performance Lab plugin. We shouldn't have to replace images that are controlled by code, we should return the right images from the very beginning. And unfortunately this isn't something we can do from within the plugin, most importantly because we cannot add the parameter to the core function from here. If anything, we would need to rewrite the whole function which I also think is not worth it.

We are adding the necessary infrastructure pieces in the plugin, but for anything beyond images in user-generated content, we'll have to wait for the core implementation.

In other words, I agree we should address this feature request, but not in the plugin implementation. Adding the parameter and updating the core default is already on the radar.
